x/gov/keeper/msg_server.go

```diff
@@ -85,6 +111,10 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1.MsgSubmitPropos
 
 // CancelProposals implements the MsgServer.CancelProposal method.
 func (k msgServer) CancelProposal(goCtx context.Context, msg *v1.MsgCancelProposal) (*v1.MsgCancelProposalResponse, error) {
+	if _, err := sdk.AccAddressFromBech32(msg.Proposer); err != nil {
```

Comment:
This is already checked l119. We should move that up here, and update the error message of l119 to this one.